### PR TITLE
chore(istio): add version metadata to binaries

### PR DESCRIPTION
**Pull request by stormqueen1990**
Ensure version metadata is added to the binaries.

### CODE REVIEW

File: istio-pilot-agent-1.22.yaml [hunk 1]
```diff
@@ -30,7 +30,13 @@ pipeline:
...
       output: pilot-agent
       # istio default ldflags: https://github.com/istio/istio/blob/master/Makefile.core.mk#L210
-      ldflags: "-extldflags -static -s -w"
+      # Extracted from https://github.com/istio/istio/blob/4358b84b911a80ba09ef36ac00ad85535a77e7ca/common/scripts/report_build_info.sh#L41-L48
+      ldflags: |
+        -extldflags -static -s -w
+        -X istio.io/istio/pkg/version.buildVersion=${{package.version}}
+        -X istio.io/istio/pkg/version.buildGitRevision=$(git rev-parse HEAD)
+        -X istio.io/istio/pkg/version.buildTag=$(git describe --tags --always)
+        -X istio.io/istio/pkg/version.buildStatus="$(if git diff-index --quiet HEAD --; then echo "Clean"; else echo "Modified"; fi)"
```
Comment (joshrwolf) [inline istio-pilot-agent-1.22.yaml:39]:
```suggestion
        -X istio.io/istio/pkg/version.buildStatus="Clean"
```

I think we just want to override this to always be "Clean" given how our build system and `git-checkout` pipelines work, but it'd be worth checking if this has any sideeffects other than how its displayed in the binary

File: istio-pilot-agent-1.22.yaml [hunk 2]
```diff
@@ -30,7 +30,13 @@ pipeline:
       packages: ./pilot/cmd/pilot-agent
       output: pilot-agent
       # istio default ldflags: https://github.com/istio/istio/blob/master/Makefile.core.mk#L210
-      ldflags: "-extldflags -static -s -w"
+      # Extracted from https://github.com/istio/istio/blob/4358b84b911a80ba09ef36ac00ad85535a77e7ca/common/scripts/report_build_info.sh#L41-L48
+      ldflags: |
+        -extldflags -static -s -w
```
Comment (joshrwolf) [inline istio-pilot-agent-1.22.yaml:35]:
```suggestion
        -extldflags -s -w
```

File: istio-pilot-agent-1.22.yaml (hunk 1 above)
Comment (stormqueen1990) [reply istio-pilot-agent-1.22.yaml]: Updated!

File: istio-pilot-agent-1.22.yaml (hunk 2 above)
Comment (stormqueen1990) [reply istio-pilot-agent-1.22.yaml]: Removed!

**Review by joshrwolf — APPROVED** [review]

**Pull request merged**